lib/sycamore/sycamore/materialize.py

```diff
@@ -328,24 +338,29 @@ def before(node):
                 return node
 
             path = self._path.copy()
-            path["root"] = self.directory / materialize["name"]
+            path["root"] = self._directory / materialize["name"]
             materialize["mark"] = True
             materialize["count"] = materialize.get("count", 0) + 1
             return Materialize(node, context, path=path)
 
         return before
 
-    def _setup_directory(self):
+    def _choose_directory(self, path):
         from pathlib import Path
 
-        if self.directory is None:
+        directory = path.pop("root", None)
+        self._path = path
+
+        if directory is None:
             from datetime import datetime
             import tempfile
 
             now = datetime.now().replace(microsecond=0)
             dir = Path(tempfile.gettempdir()) / f"materialize.{now.isoformat()}"
```

Comment:
does this ever leak storage?

Reply:
It's intended to "leak". You're asking it to materialize so it's trying to leave it around.